core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -85,6 +85,8 @@ public void accept(String file) {
   private ExecutorService planExecutorService = ThreadPools.getWorkerPool();
   private Boolean incrementalCleanup;
 
+  private List<Snapshot> expiredSnapshots;
```

Comment:
I think it would be better to keep a Set of snapshot IDs, because removing from a list is O(n)